src/unistd.rs

```diff
@@ -11,18 +11,126 @@ use std::os::unix::io::RawFd;
 use std::path::{PathBuf};
 use void::Void;
 use sys::stat::Mode;
+use std::fmt;
 
 #[cfg(any(target_os = "linux", target_os = "android"))]
 pub use self::linux::*;
 
+/// User identifier
+///
+/// Newtype pattern around `uid_t` (which is just alias). It prevents bugs caused by accidentally
+/// passing wrong value.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub struct Uid(uid_t);
+
+impl Uid {
+    pub fn from_raw(uid: uid_t) -> Self {
+        Uid(uid)
+    }
+
+    pub fn current() -> Self {
+        getuid()
+    }
+
+    pub fn effective() -> Self {
+        geteuid()
+    }
+
+    pub fn is_root(self) -> bool {
```

Comment:
I'm not aware of any downsides (apart from theoretical slow down, which should be optimized-out). I implemented it this way mostly because I didn't care, since it's `Copy`. I understand the style and, I'm going to change it.